internal/config/options.go

```diff
@@ -84,12 +84,17 @@ func WithLogger(logger logr.Logger) Option {
 		o.logger = logger
 	}
 }
+func WithOnChangeHandler(h ChangeHandler) Option {
```

Comment:
Do we need to expose `ChangeHandler`? 

Would it be possible to keep `ChangeHandler` package private and just expose a method to register a callback?

Reply:
Sure it works both.